cmd/minishift/cmd/profile/profile_copy.go

```diff
@@ -0,0 +1,142 @@
+	cmdUtil.CreateMinishiftDirs(newInstanceDirs)
+
+	// Write the config information to the new configfile
+	targetConfFile := constants.GetProfileConfigFile(profileName)
+
+	//Create the config.json for the new profile
+	cmdUtil.EnsureConfigFileExists(targetConfFile)
+
+	// Check if the config file for the new profile is not empty
+	targetConfig, err := config.ReadViperConfig(targetConfFile)
+	if err != nil {
+		atexit.ExitWithMessage(1, err.Error())
+	}
+	if len(targetConfig) != 0 {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Can not copy configurations from '%s' to '%s' as '%s' is not empty", srcProfile, profileName, targetConfFile))
+	}
+
+	err = config.WriteViperConfig(targetConfFile, sourceConfig)
+	if err != nil {
+		atexit.ExitWithMessage(1, err.Error())
+	}
+
+	//Get directory structure of the source profile
+	srcProfileDirs := cmdState.GetMinishiftDirsStructure(constants.GetProfileHomeDir(srcProfile))
+
+	// Copy addons from source profile to new profile
+	// Removing the newly created addon directory as filehelper.CopyDir does not expect the directory to be present.
+	err = os.Remove(newInstanceDirs.Addons)
+	if err != nil {
+		atexit.ExitWithMessage(1, err.Error())
+	}
+
+	err = filehelper.CopyDir(srcProfileDirs.Addons, newInstanceDirs.Addons)
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Copying add-ons to profile '%s' failed : %s", profileName, err.Error()))
+	}
+
+	// Get the machine config file (MACHINE_NAME.json) of the target profile
+	targetProfileInstanceConfig, err := config.NewInstanceConfig(minishiftConstants.GetProfileInstanceConfigPath(profileName))
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating config for profile '%s' : %s", profileName, err.Error()))
+	}
+
+	//Get the instance machine config of the source profile
+	srcProfileInstanceConfig, err := config.NewInstanceConfig(minishiftConstants.GetProfileInstanceConfigPath(srcProfile))
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating config for source profile '%s' : %s", srcProfile, err.Error()))
+	}
+
+	// Copy addon state information from config/minishift.json
+	targetProfileInstanceConfig.AddonConfig = srcProfileInstanceConfig.AddonConfig
+	if err != targetProfileInstanceConfig.Write() {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error copying add-on config to profile '%s': %s", profileName, err.Error()))
+	}
+
+	fmt.Println(fmt.Sprintf("Profile '%s' is created successfully using configs from profile '%s'", profileName, srcProfile))
+}
+
+func validateCopyProfileCmd(args []string) {
+	missingArgumentMessage := "Source and a target profile (new profile) name must be provided. Run 'minishift profile list' for a list of existing profiles."
+	invalidNameMessage := "Profile names must consist of alphanumeric characters only."
+	if len(args) < 2 {
+		atexit.ExitWithMessage(1, missingArgumentMessage)
+	} else if strings.TrimSpace(args[0]) == "" {
```

Comment:
I don't think this block is needed since if you put spaces only that will automatic ignored and your len(args) would be less than 2.